posts/tips-and-tricks-for-manually-scaling-a-global-dynamodb-table-from-an-aws-lambda.md

```diff
@@ -0,0 +1,128 @@
+}
+```
+
+# Full Version
+Scale a global table
+- In this example, the primary table is in _us-east-1_. A global table was created with a replica in _us-west-2_
+- Global tables must use auto-scaling
+- Setting the RCU/WCU explicitly on an auto-scaled table does not actually change the RCU/WCU (approach for MVP will not work)
+- Upgrade to SDK v2 to simplify setting the auto-scaling parameters
+- Set the minimum RCU/WCU on the auto-scaler instead
+- Initialize
+```java
+private DynamoDbClient getClient() {
+    ApacheHttpClient.Builder httpClientBuilder = ApacheHttpClient.builder()
+            .connectionTimeout(CONNECTION_TIMEOUT)
+            .socketTimeout(SOCKET_TIMEOUT);
+    RetryPolicy.Builder retryBuilder = RetryPolicy.builder()
+            .numRetries(RETRY_LIMIT);
+    ClientOverrideConfiguration.Builder clientOverrideBuilder =
+            ClientOverrideConfiguration.builder()
+                    .apiCallAttemptTimeout(API_CALL_ATTEMPT_TIMEOUT)
+                    .apiCallTimeout(API_CALL_TIMEOUT)
+                    .retryPolicy(retryBuilder.build());
+
+    return DynamoDbClient.builder()
+            .httpClientBuilder(httpClientBuilder)
+            .overrideConfiguration(clientOverrideBuilder.build())
+            .build();
+}
+```
+- Call the SDK
+```java
+AutoScalingTargetTrackingScalingPolicyConfigurationUpdate policyConfigurationUpdate =
+        AutoScalingTargetTrackingScalingPolicyConfigurationUpdate.builder()
+                .targetValue(SCALING_POLICY_TARGET_VALUE)
+                .build();
+AutoScalingPolicyUpdate policyUpdate = AutoScalingPolicyUpdate.builder()
+        .targetTrackingScalingPolicyConfiguration(policyConfigurationUpdate)
+        .build();
+AutoScalingSettingsUpdate.Builder updateBuilder = AutoScalingSettingsUpdate.builder()
+        .maximumUnits(MAX_AUTOSCALE_CAPACITY)
+        .scalingPolicyUpdate(policyUpdate);
+
+AutoScalingSettingsUpdate rcuUpdate = updateBuilder.minimumUnits(HIGH_READ_CAPACITY_UNITS).build();
+AutoScalingSettingsUpdate wcuUpdate = updateBuilder.minimumUnits(HIGH_WRITE_CAPACITY_UNITS).build();
+
+ReplicaAutoScalingUpdate eastReplicaUpdate = ReplicaAutoScalingUpdate.builder()
+        .regionName("us-east-1")
+        .replicaProvisionedReadCapacityAutoScalingUpdate(rcuUpdate)
+        .build();
+ReplicaAutoScalingUpdate westReplicaUpdate = ReplicaAutoScalingUpdate.builder()
+        .regionName("us-west-2")
+        .replicaProvisionedReadCapacityAutoScalingUpdate(rcuUpdate)
+        .build();
+UpdateTableReplicaAutoScalingRequest autoScalingRequest = UpdateTableReplicaAutoScalingRequest.builder()
+        .tableName(TABLE_NAME)
+        .replicaUpdates(eastReplicaUpdate, westReplicaUpdate)
+        .provisionedWriteCapacityAutoScalingUpdate(wcuUpdate)
+        .build();
+
+try {
+    getClient().updateTableReplicaAutoScaling(autoScalingRequest);
+} catch (AmazonServiceException e) {
+    System.err.println(e.getErrorMessage());
+    System.exit(1);
+}
+```
+
+# Issues
```

Comment:
I like this section and wish more technical posts had this.

Reply:
Thanks